### PR TITLE
feat: run nested async workflows

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**

### CODE REVIEW

File: packages/core/orchestration/src/transaction/transaction-orchestrator.ts [hunk 1]
```diff
@@ -783,6 +783,10 @@ export class TransactionOrchestrator extends EventEmitter {
               })
           )
         } else {
+          console.log(flow.modelId, "BEGIN EXECUTION", {
+            metadata: flow.metadata,
+          })
```
Comment (adrien2p) [inline packages/core/orchestration/src/transaction/transaction-orchestrator.ts]: todo: rm :)

**Review by adrien2p — APPROVED** [review]

**Comment by adrien2p** [comment]
should we add some tests to validate the event releasing with nested workflows?

File: packages/modules/workflow-engine-redis/integration-tests/__tests__/index.spec.ts [hunk 2]
```diff
@@ -35,10 +35,9 @@ moduleIntegrationTestRunner<IWorkflowEngineService>({
   },
   testSuite: ({ service: workflowOrcModule, medusaApp }) => {
     describe("Workflow Orchestrator module", function () {
-      const afterEach_ = async () => {
+      const afterEach_ = async () => {}
```
Comment (sradevski) [inline packages/modules/workflow-engine-redis/integration-tests/__tests__/index.spec.ts]: Very NIT: We could just remove this

File: packages/modules/workflow-engine-redis/src/services/workflow-orchestrator.ts [hunk 3]
```diff
@@ -198,21 +222,35 @@ export class WorkflowOrchestratorService {
...
-    // TODO: temporary
+    const hasFinished = ret.transaction.hasFinished()
+    const metadata = ret.transaction.getFlow().metadata
+    const { parentStepIdempotencyKey } = metadata ?? {}
+    const hasFailed = [
+      TransactionState.REVERTED,
+      TransactionState.FAILED,
+    ].includes(ret.transaction.getFlow().state)
+
+    // TODO: temporary - Create an Aknowledgement type
```
Comment (sradevski) [inline packages/modules/workflow-engine-redis/src/services/workflow-orchestrator.ts]: I think Shahed already created one, but I am not 100% certain

**Review by sradevski — APPROVED** [review]
Nice!

File: packages/core/orchestration/src/transaction/types.ts [hunk 4]
```diff
@@ -237,6 +242,7 @@ export type TransactionFlow = {
   transactionId: string
   metadata?: {
     eventGroupId?: string
+    parentIdempotencyKey?: string
```
Comment (adrien2p) [inline packages/core/orchestration/src/transaction/types.ts:245]: suggestion: should we store the parent workflow id, I am thinking for example for the UI and for the telemetry that way we don't have to infer it from the idempotency key which in theory can be provided by the consumer. does that make sense?

**Review by adrien2p — APPROVED** [review]
LGTM, we will just have to keep an eye on bull :D

**Comment by adrien2p** [comment]
This will probably fix import/export as well as I just saw that it was relying on async run as step inside a non async workflow 😂 cc @sradevski I think we have been lucky with this one right? (do we test non happy path for it?)

**Comment by sradevski** [comment]
@adrien2p yeah could be that we haven't caught it, there are few tests but it's not extensively tested.

File: packages/core/orchestration/src/transaction/types.ts (hunk 4 above)
Comment (carlos-r-l-rodrigues) [reply packages/core/orchestration/src/transaction/types.ts:245]: I will let this for another iteration where we can potentially have it as an array.
I believe when we focus in the UI we can tackle this and decide which approach is better.

**Pull request merged**